sources/logdog/src/create_tarball.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use std::fs::File;
+use std::path::PathBuf;
+
+use flate2::write::GzEncoder;
+use flate2::Compression;
+use snafu::ResultExt;
+use tar;
+
+// Creates a tarball with all the contents of directory `dir`.
+// Outputs as file to `outfile`.
+pub(crate) fn create_tarball(dir: &PathBuf, outfile: &PathBuf) -> crate::error::Result<()> {
```

Comment:
It's better to accept anything like a path here, if you don't need ownership:
```suggestion
pub(crate) fn create_tarball(dir: P1, outfile: P2) -> crate::error::Result<()>
where P1: AsRef<Path>,
   P2: AsRef<Path>,
{
```
(P1 and P2 so the input types don't have to be the same concrete type)

Then, if you're using the path a lot, you can do a `let dir = dir.as_ref();` at the top instead of calling `.as_ref()` everywhere.

Same thing with string-like things (`AsRef<str>`).

Would you be willing to update this elsewhere too?